### ISSUE TITLE
cant load Format.php library thru composer

### ISSUE THREAD

**Issue by detorresvc** [chriskacerguis/codeigniter-restserver#859]
in REST_Controller.php 

use Exception;
use stdClass;
use Restserver\Libraries\Format;
+require('Format.php');

to use the library

**Comment by Hanisch-IT** [comment]
You dont need `use`, you dont need `require`. The rest controller and format are in the global context `/`.

Only requirement ist to configurate your composer autoload file in config.php

**Comment by peterson-umoke** [comment]
still doesnt work, even by the way the config files are missing in the composer's version, only the github version is up to date...

**Comment by farukgaric** [comment]
Config files are coming with restserver as dependency but you should overwrite it if you want change some configuration params. You can overwrite them if you copy rest.php to your config directory. 
Composer is using github to get restserver.

**Comment by Hanisch-IT** [comment]
@peterson-umoke you have standard config files in composer (same as in github version) that will be loaded. If you want to overwrite configs, its possible to create your own `rest.php` in your config folder and set the configurations there.

See: https://github.com/chriskacerguis/codeigniter-restserver#configuration

**Comment by antwal** [comment]
Hi 

same problem, 

if install library with composer it not working with autoload namespaces, pls fix it

on autoload_classmap i see this: 
`'Format' => $vendorDir . '/chriskacerguis/codeigniter-restserver/application/libraries/Format.php',`
`'Restserver\\Libraries\\REST_Controller' => $vendorDir . '/chriskacerguis/codeigniter-restserver/application/libraries/REST_Controller.php',`

if need use old method with php require, it is useless to migrate on composer

Thanks.

**Comment by Hanisch-IT** [comment]
Which Restserver version do you have? The REST_Controller does not have a namespace.
So it should not get autoloaded by namespace

**Comment by chriskacerguis** [comment]
no response.  closing.

**Comment by takkyuuplayer** [comment]
> Which Restserver version do you have? The REST_Controller does not have a namespace.
> https://github.com/chriskacerguis/codeigniter-restserver/issues/859#issuecomment-339416566

I have the same issue on https://github.com/takkyuuplayer/php-code_igniter-exercise/pull/5/files
Reseserver's version is 3.0.0 and my autoload_classmap is

```
'Format' => $vendorDir . '/chriskacerguis/codeigniter-restserver/application/libraries/Format.php',
'Restserver\\Libraries\\REST_Controller' => $vendorDir . '/chriskacerguis/codeigniter-restserver/application/libraries/REST_Controller.php',
```

### How to reproduce the problem.

1. Boot docker

    ```
    $ git clone https://github.com/takkyuuplayer/php-code_igniter-exercise.git
    $ cd php-code_igniter-exercise
    $ git checkout -t origin/rest-api
    $ cd docker
    $ make up
    $ make attach
    ```

2. Then in docker

    ```
    $ make
    $ php index.php rest
    PHP Fatal error:  Class 'REST_Controller' not found in /var/www/application/controllers/Rest.php on line 5

    Fatal error: Class 'REST_Controller' not found in /var/www/application/controllers/Rest.php on line 5

    A PHP Error was encountered

    Severity:    Error
    Message:     Class 'REST_Controller' not found
    Filename:    /var/www/application/controllers/Rest.php
    Line Number: 5

    Backtrace:
    ```

**Comment by farukgaric** [comment]
@takkyuuplayer You have to use at least 3.0.1 if you want composer support. It's always better to use latest version.

Try change restserver's version and update composer to see is it working. If not, we can investigate further.

**Comment by takkyuuplayer** [comment]
Thank you. I noticed since I named my repository in composer.json wrongly (I left as it as default when I installed codeigniter),  `/composer.phar require chriskacerguis/codeigniter-restserver` couldn't resolve its dependency to install the latest version.

https://github.com/takkyuuplayer/php-code_igniter-exercise/pull/5/commits/22880fdab7b7be6871eea9cee8bda99d760fad2d

Now I can install the latest version via composer.
Thanks.
